libfreerdp/core/capabilities.c

```diff
@@ -195,12 +180,54 @@ static BOOL rdp_read_general_capability_set(wStream* s, rdpSettings* settings)
 		 * server-only flags indicating to the client weather the
 		 * respective PDUs are supported. See MS-RDPBCGR 2.2.7.1.1
 		 */
-		if (!refreshRectSupport)
+		if (!src->RefreshRect)
 			settings->RefreshRect = FALSE;
 
-		if (!suppressOutputSupport)
+		if (!src->SuppressOutput)
 			settings->SuppressOutput = FALSE;
 	}
+	return TRUE;
+}
+
+/**
+ * Read general capability set.\n
+ * @msdn{cc240549}
+ * @param s stream
+ * @param settings settings
+ * @return if the operation completed successfully
+ */
+
+static BOOL rdp_read_general_capability_set(wStream* s, rdpSettings* settings)
+{
+	UINT16 extraFlags;
+	BYTE refreshRectSupport;
+	BYTE suppressOutputSupport;
+
+	if (!Stream_CheckAndLogRequiredLength(TAG, s, 20))
```

Comment:
WINPR_ASSERT missing for settings and s

Reply:
`s` is asserted by `Stream_CheckAndLogRequiredLength`